### PR TITLE
Update examples for Auth Next.

### PR DESCRIPTION
**Pull request by dmkozh**
### What

- Introduced 3 new examples
- Removed redundant examples
- Updated most of the examples that needed authorization to use Auth Next

### Why

Support the latest SDK version

### Known limitations

There are still 2-3 more complex examples that need to be updated (token, liquidity pool and maybe lp router, though it probably can be removed)

<a href="https://gitpod.io/#https://github.com/stellar/soroban-examples/pull/204"></a>

### CODE REVIEW

File: single_offer/src/lib.rs [hunk 1]
```diff
@@ -13,202 +12,146 @@ mod token {
...
+        let contract = e.current_contract_address();
+
+        // Perform the trade in 3 `xfer` steps.
+        // Note, that we don't need to verify any balances - the contract would
+        // just trap and roll back in case if any of the transfers fails for
+        // any reason, including insufficient balance.
+
+        // Transfer the `buy_token` from buyer to this contract.
+        // This `xfer` call should be authorized by buyer.
+        // This could as well be a direct transfer to the seller, but sending to
```
Comment (sisuresh) [inline single_offer/src/lib.rs:108]: Should we encourage this? This contract would be cheaper to use with one less xfer.

File: single_offer/src/lib.rs [hunk 2]
```diff
@@ -13,202 +12,146 @@ mod token {
...
-
-        transfer_sell(&e, to, amount);
-
-        let admin = read_administrator(&e);
-        transfer_buy(&e, admin, balance_buy_token);
+    // Trades `buy_token_amount` of buy_token from buyer for `sell_token` amount
+    // defined by the price.
+    // Buyer needs to authorize the `trade` call and internal `xfer` call to
+    // the contract address.
+    pub fn trade(e: Env, buyer: Address, buy_token_amount: i128) {
```
Comment (sisuresh) [inline single_offer/src/lib.rs:84]: I see you removed the min logic. Was that intentional?

File: single_offer/src/lib.rs (hunk 1 above)
Comment (dmkozh) [reply single_offer/src/lib.rs:108]: That's true, but in a general case it's probably hard to achieve (and I really believe that limiting the scope of trust would have a serious UX benefit). For the next release we can address this with something like `xfer_contract` we've discussed at some point, which basically would fuse allowance+xfer_from into a single call (sign the max_amount+contract, then use actual amount/actual recipient at execution time).

File: single_offer/src/lib.rs (hunk 2 above)
Comment (dmkozh) [reply single_offer/src/lib.rs:84]: I thought it's unnecessary, but now I realize that it's possible to frontrun a trade with a price update, which is bad. I'll commit a fix soon.

File: single_offer/src/lib.rs (hunk 1 above)
Comment (sisuresh) [reply single_offer/src/lib.rs:108]: Yeah that's a valid point. I was thinking that that the call stack should be sufficient for trust, but it still may not be from the users perspective

**Review by sisuresh — APPROVED** [review]

**Pull request merged**